### ISSUE TITLE
potential race condition in program initialization

### ISSUE THREAD

**Issue by jleightcap** [accessibility-oscilloscope/buildroot#10]
our programs look something like

```c
// program p1.
int main(void) {
   mkfifo("p1-fifo");
   open("p1-fifo", O_RDONLY); // fine

   open("p2-fifo", O_WRONLY); // does that FIFO exist yet?
}
```

the `open` on a fifo that has not yet been created blocks, and does not unblock when the FIFO is subsequently made.

the init script forks the programs in reverse order, such that a FIFO _might_ be created before it's used.

**Comment by jleightcap** [comment]
potential solutions, or at least seeing if this is a problem:

- make sure to use syslog consistently for each return value of `open`
- factor out all FIFO creation to before all program initialization, so those files are guaranteed to exists.
- nothing goes wrong and pretend it never happened

**Comment by cnorthway** [comment]
fixed in 2396c31d7a8d3af9dce269213c8f1af459185a87, factored out FIFO creation into a separate step in init
